src/providers/retryProvider.ts

```diff
@@ -88,9 +89,27 @@ export class RetryProvider extends ethers.providers.StaticJsonRpcProvider {
     ): Promise<[ethers.providers.StaticJsonRpcProvider, unknown]> => {
       return this._trySend(provider, method, params)
         .then((result): [ethers.providers.StaticJsonRpcProvider, unknown] => [provider, result])
-        .catch((err) => {
+        .catch((err: unknown) => {
+          if (JsonRpcError.is(err)) {
+            const {
+              error: { code, message },
+            } = err;
+            // [-32768, -32100] is reserved by the JSON-RPC spec.
+            // [-32099, -32000] is allocated for implementation-defined responses.
+            // Everything else is available for use by the application space.
+            // Most node implementations return 3 for an eth_call revert, but some return -32000.
+            // See also https://www.jsonrpc.org/specification
+            if (code < -32768 || code > -32100) {
+              if (method === "eth_call" && message.toLowerCase().includes("revert")) {
```

Comment:
This could maybe also be conditional on `quorumThreshold === 1`.

Reply:
https://github.com/across-protocol/sdk/pull/757/commits/90374cb8545867b7c70d2cf031c80a4d58fbb006